src/partition/growfs.c

```diff
@@ -16,13 +18,35 @@
 #include "path-util.h"
 #include "strv.h"
 
+static int resize_ext4(int mountfd, int devfd, uint64_t numblocks, int blocksize) {
+        if (ioctl(mountfd, EXT4_IOC_RESIZE_FS, &numblocks) != 0)
+                return -errno;
+
+        return 0;
+}
+
+static int resize_btrfs(int mountfd, int devfd, uint64_t numblocks, int blocksize) {
```

Comment:
maybe move this into btrfs-util.[ch]? There are other invocations of that ioct there already

Reply:
I'm still trying to figure out what information is necessary for which fs. Ext4 requires the size in blocks, btrfs is happy with "max"... Let's keep things close for now, because it's easier to edit, and maybe move them at the end depending on how things look.